pstoremgr/pstoremgr.go

```
@@ -243,41 +244,50 @@ func (pm *Manager) LoadPeerstore() (addrs []ma.Multiaddr) {
 
 // SavePeerstore stores a slice of multiaddresses in the peerstore file, one
 // per line.
-func (pm *Manager) SavePeerstore(pinfos []peer.AddrInfo) {
+func (pm *Manager) SavePeerstore(pinfos []peer.AddrInfo) error {
 	if pm.peerstorePath == "" {
-		return
+		err := errors.New("empty peerstore path")
+		logger.Error(err)
+		return err
 	}
 
 	pm.peerstoreLock.Lock()
 	defer pm.peerstoreLock.Unlock()
 
 	f, err := os.Create(pm.peerstorePath)
 	if err != nil {
-		logger.Warningf(
+		logger.Errorf(
 			"could not save peer addresses to %s: %s",
 			pm.peerstorePath,
 			err,
 		)
-		return
+		return err
 	}
 	defer f.Close()
 
 	for _, pinfo := range pinfos {
+		if len(pinfo.Addrs) == 0 {
+			err = errors.New("address info does not have any multiaddresses")
+			logger.Error(err)
+			return err
+		}
+
 		addrs, err := peer.AddrInfoToP2pAddrs(&pinfo)
 		if err != nil {
-			logger.Warning(err)
-			continue
+			logger.Error(err)
+			return err
 		}
 		for _, a := range addrs {
 			f.Write([]byte(fmt.Sprintf("%s\n", a.String())))
```

HERE, here is where you want to return an error if writing to the file failed